File: src/examples/bcast_reduce.c
#include "mpi.h"

main(int argc, char** argv){

  int my_rank, s=0, x=0;

  MPI_Init(&argc, &argv);
  //MPI_Init(NULL, NULL);
  MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);

  if(my_rank==0) x=2;

  MPI_Bcast(&x, 1, MPI_INT, 0, MPI_COMM_WORLD);

  x *= my_rank;

  MPI_Reduce(&x, &s, 1, MPI_INT, MPI_SUM, 0, MPI_COMM_WORLD);

  if(my_rank==0) printf("The sum is %d.\n", s);

  MPI_Finalize();

}

File: src/examples/bcast_reduce.f90
program main

  implicit none
  include 'mpif.h'

  integer errcode, my_rank, s, x

  call MPI_INIT(errcode)
  call MPI_COMM_RANK(MPI_COMM_WORLD, my_rank, errcode)

  if(my_rank==0) x=2 
  !if(my_rank==0) then
  !   x=2
  !endif

  call MPI_Bcast(x, 1, MPI_INT, 0, MPI_COMM_WORLD, errcode)

  x = x * my_rank

  call MPI_Reduce(x, s, 1, MPI_INT, MPI_SUM, 0, MPI_COMM_WORLD, errcode)

  if(my_rank==0) print *, 'The sum is:', s

  call MPI_FINALIZE(errcode)

end program main

